sqe-database-access/Models/SignInterpretation.cs

```diff
@@ -13,7 +13,22 @@ public class SignInterpretation
 
     public class NextSignInterpretation
     {
-        public uint nextSignInterpretationId { get; set; }
-        public uint signSequenceAuthor { get; set; }
+        public uint nextSignInterpretationId { get; }
+        public uint signSequenceAuthor { get; }
+        
+        // The override for Equals and GetHashCode methods here enable the
+        // HashSet nextSignInterpretations of the SignInterpretation object
+        // to ensure that no duplicate values will be inserted into the set.
+        public override bool Equals(object obj)
+        {
+            return obj is NextSignInterpretation q 
```

Comment:
This `is` operator usage is new to me. I actually don't like it very much, I wonder why they've added it.